### ISSUE TITLE
AbstractTypeMaterializer attempts to materialize Iterable

### ISSUE THREAD

**Issue by jmax01** [FasterXML/jackson-module-mrbean#28]
AbstractTypeMaterializer should skip Iterable.class.

Something like this perhaps?

``` java
        //current
        if (type.isContainerType() || type.isPrimitive() || type.isEnumType() || type.isThrowable() ||
                type.getRawClass() == Number.class)
        {
            return null;
        }
       Class<?> cls = type.getRawClass();

        //fix
        Class<?> cls = type.getRawClass();
        if (type.isContainerType() || type.isPrimitive() || type.isEnumType() || type.isThrowable() ||
                cls  == Number.class || cls == Iterable.class)
        {
            return null;
        }       
```

**Comment by georgschmidtdumont** [comment]
Looking at the code this seems to be fixed as of now? (The code comment even says June 2016)
We're currently facing a similar issue, where we're having a `Map<String, Serializable>` and an attempt is made to materialize the `Serializable` interface. Probably this type should be excluded as well?

**Comment by cowtowncoder** [comment]
@georgschmidtdumont I think you are right. Could you please file a separate issue for `Serializable` and I can fix that -- but it's easier to keep release notes straight when each issue covers separate problem (even if related and similar)?

I'll close this one and guesstimate it was fixed in `2.8.0`.
